grafana/resource_datasource_permission_test.go

```diff
@@ -72,10 +70,8 @@ func testAccDatasourcePermissionCheckDestroy(datasourceID *int64) resource.TestC
 		if err != nil {
 			return fmt.Errorf("error getting datasource permissions %d: %s", *datasourceID, err)
 		}
-		if response.Enabled {
-			return fmt.Errorf("datasource permissions %d still enabled", *datasourceID)
-		}
-		if len(response.Permissions) > 0 {
+		// Data source permissions for Admins are always set and can't be removed through provisioning
+		if len(response.Permissions) > 1 || (len(response.Permissions) == 1 && response.Permissions[0].BuiltInRole != "Admin") {
```

Comment:
With enterprise the admin role could be missing if the basic role admin is modified to not have required actions

Reply:
This particular change is only for tests, so shouldn't make a difference.

But more generally it is a problem, and I don't really know how to handle it. I think similar issues would arise if DS permissions were granted to viewers/editors/specific teams or users - if Terraform doesn't reflect this, it'll attempt to remove the changes but will fail at it. I'm not sure how such cases are usually handled through Terraform 🤔

Reply:
One option would be to require that the users always specify the permissions that can't be changed in their Terraform configs - here it'd mean always including the following (note that Terraform doesn't yet support `Edit` permission):
`
  permissions {
    built_in_role = "Admin"
    permission    = "Query"
  }
`